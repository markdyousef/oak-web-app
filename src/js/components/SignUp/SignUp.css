:root {
    --primary-color: #5ed892;
    --main-black-color: #131517;
    --light-black-color: #2d3438;
    --dark-gray-color: #737a80;
    --main-gray-color: #f2f2f2;
    --light-gray-color: #fafafa;
    --main-white-color: #fff;
}

.container {
    width: 100%;
    height: 100%;
    position: absolute;
    display: flex;
    align-items: center;
    justify-content: center;
}

.signup {
    width: 400px;
    background-color: var(--main-white-color);
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
    padding: 20px;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.signup h1 {
    font-size: 18px;
    margin-top: 20px;
    font-weight: bold;
}

.inputs {
    margin-top: 20px;
    width: 100%;
    display: flex;
    flex-direction: column;
    height: 120px;
    justify-content: space-between;
    padding: 5px;
}

.buttons {
    margin-top: 15px;
    height: 50px;
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.message {
    margin-bottom: 30px;
    height: 30px;
    width: 100%;
    padding: 10px;
}

.message div {
    width: 100%;
    background-color: rgba(242, 35, 35, 0.8);
    height: 100%;
    border-radius: 3px;
    display: flex;
    align-items: center;
    padding: 20px;
}

.message h5 {
    color: var(--main-white-color);
    font-size: 14px;
}
