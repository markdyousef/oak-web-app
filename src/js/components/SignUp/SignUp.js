// @flow
import React, { Component, PropTypes } from 'react';
import Input from '../shared/Input';
import Button from '../shared/Button';

import css from './SignUp.css';

class SignUp extends Component {
    static propTypes = {
        createUser: PropTypes.func.isRequired,
        router: PropTypes.shape({
            push: PropTypes.func
        }).isRequired
    };
    constructor() {
        super();
        this.onSubmit = this.onSubmit.bind(this);
        this.state = {
            name: '',
            email: '',
            password: '',
            message: null
        };
    }
    onSubmit() {
        const { createUser } = this.props;
        const { name, email, password } = this.state;

        // clear messsage
        this.setState({ message: null });

        // name validation
        if (name.length < 2) {
            return this.setState({ message: 'please provide a valid name'});
        }
        // email validation
        if (email.length < 6) {
            return this.setState({ message: 'please provide a valid email'});
        }
        // password validation
        if (password.length < 6) {
            return this.setState({ message: 'your password should be at least 6 characters' });
        }
        return createUser(name, email, password)
            .then(() => this.setState({ message: { type: 'success', text: 'Your account has been created ' } }))
            .catch(err => this.setState({ message: { type: 'failed', text: 'Lame!' } }));
    }
    render() {
        const { name, email, password, message } = this.state;
        const { router } = this.props;
        return (
            <div className={css.container}>
                <div className={css.signup}>
                    <h1>Create Account</h1>
                    <div className={css.inputs}>
                        <Input
                            title="Name"
                            value={name}
                            onChange={value => this.setState({ name: value, message: null })}
                            type="name"
                            placeholder="Name"
                        />
                        <Input
                            title="Email"
                            value={email}
                            onChange={value => this.setState({ email: value, message: null })}
                            type="email"
                            placeholder="Email"
                        />
                        <Input
                            title="Password"
                            value={password}
                            onChange={value => this.setState({ password: value, message: null })}
                            type="password"
                            placeholder="Password"
                        />
                    </div>
                    <div className={css.buttons}>
                        <Button
                            onClick={() => router.push('/')}
                            text="Login"
                            // type="transparent"
                        />
                        <Button
                            onClick={this.onSubmit}
                            text="Create Account"
                            type="primary"
                        />
                    </div>
                    <div className={css.message}>
                        {message &&
                            <div className={(message.type === 'success') ? css.success : null}>
                                <h5>{message.text}</h5>
                            </div>
                        }
                    </div>
                </div>
            </div>
        );
    }
}

export default SignUp;
