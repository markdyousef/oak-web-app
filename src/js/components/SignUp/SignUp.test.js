import React from 'react';
import renderer from 'react-test-renderer';
import SignUp from './SignUp';

it('renders correctly', () => {
    const component = renderer.create(
        <SignUp />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
