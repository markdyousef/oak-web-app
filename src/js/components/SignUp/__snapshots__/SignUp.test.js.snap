exports[`test renders correctly 1`] = `
<div
  className={undefined}>
  <div
    className={undefined}>
    <h1>
      Create Account
    </h1>
    <div
      className={undefined}>
      <div>
        <h5
          className="eKaiQM">
          Name
        </h5>
        <input
          className="bwXyts"
          onChange={[Function]}
          placeholder="Name"
          type="name"
          value="" />
      </div>
      <div>
        <h5
          className="eKaiQM">
          Email
        </h5>
        <input
          className="bwXyts"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value="" />
      </div>
      <div>
        <h5
          className="eKaiQM">
          Password
        </h5>
        <input
          className="bwXyts"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value="" />
      </div>
    </div>
    <div
      className={undefined}>
      <button
        className="bBPMRg"
        onClick={[Function]}
        type={null}>
        Login
      </button>
      <button
        className="iZsHgX"
        onClick={[Function]}
        type="primary">
        Create Account
      </button>
    </div>
    <div
      className={undefined} />
  </div>
</div>
`;
